packages/replay/src/utils.ts

```diff
@@ -88,12 +93,64 @@ export async function exponentialBackoffRetry<T>(
       if (currentAttempt == MAX_ATTEMPTS) {
         throw e;
       }
-      await waitForTime(backoff(currentAttempt));
+      await waitForTime(backOffStrategy(currentAttempt));
     }
   }
   throw Error("ShouldBeUnreachable");
 }
 
+export async function exponentialBackoffRetry<T>(
+  fn: () => Promise<T>,
+  onFail?: (e: unknown) => void
+): Promise<T> {
+  return retry(fn, geometricBackoff, onFail);
+}
+
+export async function linearBackoffRetry<T>(
+  fn: () => Promise<T>,
+  onFail?: (e: unknown) => void
+): Promise<T> {
+  return retry(fn, linearBackoff, onFail);
+}
+
+export async function concurrentWithRetry<Result>(
```

Comment:
I think `p-map` (which we're already using in this module) can do this work for you. Check out the sourcemap upload portion of `doUploadRecording` for an example